crates/pipeline_manager/src/db.rs

```diff
@@ -261,40 +321,69 @@ impl ProjectDB {
             name varchar UNIQUE NOT NULL,
             description varchar NOT NULL,
             code varchar NOT NULL,
-            schema varchar NOT NULL,
-            status varchar NOT NULL,
-            error varchar NOT NULL,
+            schema varchar,
+            status varchar,
+            error varchar,
             status_since integer NOT NULL)"#,
         )
         .execute(&conn)
         .await?;
 
+        sqlx::query(
+            r#"
+        CREATE TABLE IF NOT EXISTS pipeline (
+            id integer PRIMARY KEY AUTOINCREMENT,
+            config_id integer NOT NULL,
+            config_version integer NOT NULL,
+            -- TODO: add 'host' field when we support remote pipelines.
+            port integer,
+            killed integer NOT NULL,
+            created integer NOT NULL)"#,
+        )
+        // killed should be a boolean but the Any driver does not support it for sqlite
+        .execute(&conn)
+        .await?;
+
         sqlx::query(
             r#"
         CREATE TABLE IF NOT EXISTS project_config (
             id integer PRIMARY KEY AUTOINCREMENT,
-            project_id integer NOT NULL,
+            pipeline_id integer,
+            project_id integer,
             version integer NOT NULL,
             name varchar NOT NULL,
+            description varchar NOT NULL,
             config varchar NOT NULL,
-            FOREIGN KEY (project_id) REFERENCES project(id) ON DELETE CASCADE);"#,
+            FOREIGN KEY (project_id) REFERENCES project(id) ON DELETE CASCADE
+            FOREIGN KEY (pipeline_id) REFERENCES pipeline(id) ON DELETE SET NULL);"#,
         )
         .execute(&conn)
         .await?;
 
         sqlx::query(
             r#"
-        CREATE TABLE IF NOT EXISTS pipeline (
+        CREATE TABLE IF NOT EXISTS attached_connector (
             id integer PRIMARY KEY AUTOINCREMENT,
-            project_id integer NOT NULL,
-            project_version integer NOT NULL,
-            -- TODO: add 'host' field when we support remote pipelines.
-            port integer,
-            killed integer NOT NULL,
-            created integer NOT NULL,
-            FOREIGN KEY (project_id) REFERENCES project(id) ON DELETE CASCADE)"#,
+            uuid varchar UNIQUE NOT NULL,
+            config_id integer NOT NULL,
+            connector_id integer NOT NULL,
+            config varchar,
+            is_input boolean NOT NULL,
+            FOREIGN KEY (config_id) REFERENCES project_config(id) ON DELETE CASCADE
+            FOREIGN KEY (connector_id) REFERENCES connector(id) ON DELETE CASCADE)"#,
+        )
+        .execute(&conn)
+        .await?;
+
+        sqlx::query(
+            r#"
+            CREATE TABLE IF NOT EXISTS connector (
```

Comment:
We need to grow test coverage for the pipeline manager. Given that you've added some new tables and methods, could you also add some tests?

Reply:
I'm planning to solve test-coverage with proptests & a model; but ideally I can do it with the next PR after this one which revamps the way we do versioning (which would mean tweak many tests again anyways).

Reply:
Okay, sounds good.